src/android/Chromecast.java

```diff
@@ -609,9 +609,7 @@ public void run() {
 				List<RouteInfo> routeList = mMediaRouter.getRoutes();
 
 				for (RouteInfo route : routeList) {
-					if (!route.getName().equals("Phone") && route.getId().indexOf("Cast") > -1) {
-						sendJavascript("chrome.cast._.routeAdded(" + routeToJSON(route) + ")");
-					}
+					onRouteAdded(mMediaRouter, route);

```

Comment:
I would like you to add some of your explanations here

Reply:
change the direct call to the javascript callback "routeAdded" to the java method onRouteAdded, which in turn calls the javascript callback but also calls checkReceiverAvailable. That way if a cast device is available the javascript callback "receiverAvailable" will be called - that was not the case before and thats why often no chromecast devices were shown in the "Play On" dialog (when pressing the cast button in the web ui header)